program/src/state.rs

```diff
@@ -150,6 +150,27 @@ impl Config {
     }
 }
 
+/// The status of a governance proposal.
+#[derive(Clone, Copy, Debug, PartialEq, IntoPrimitive, TryFromPrimitive)]
+#[repr(u8)]
+pub enum ProposalStatus {
+    /// The proposal is in the draft stage.
+    Draft,
+    /// The proposal is in the voting stage.
+    Voting,
+    /// The proposal was cancelled.
+    Cancelled,
+    /// The proposal was accepted.
+    Accepted,
```

Comment:
Yeah, it's not used right now, but I thought it might be used in the near future as I add a few more things.

Technically speaking, the stage between "cooldown closed" and "processed" is a valid stage, and it's best represented by a variant like `Accepted`. We may or may not incorporate a check for `Accepted` in something like `ProcessProposal`, but even if we don't, it's a bit simpler to query accepted proposals that can be processed.

Reply:
Just adding another note here: When I've implemented vote timing (ie. voting closes after some time period), it helps to mark this `Accepted` or `Rejected` when that time expires.

Reply:
This is the one that needs this comment: In that case, let's just remove it in the meantime, that way it'll make it easier to review the PR which does add it

Reply:
I do use it in the most recent commit. Do you want me to revert part of that commit?

Reply:
Oh sorry, I kind of skipped over it because the function says it needs to be rearchitected, and I wasn't sure what that meant :sweat_smile: so do you want this to go in as is?

Reply:
Yeah I think so.

Reply:
okey-dokey, feel free to rebase and I'll take a last look!